server/src/domain/person/person.service.ts

```diff
@@ -17,10 +17,19 @@ export class PersonService {
     @Inject(IJobRepository) private jobRepository: IJobRepository,
   ) {}
 
-  async getAll(authUser: AuthUserDto): Promise<PersonResponseDto[]> {
+  async getAll(authUser: AuthUserDto, areHidden: boolean): Promise<PersonResponseDto[]> {
     const people = await this.repository.getAll(authUser.id, { minimumFaceCount: 1 });
     const named = people.filter((person) => !!person.name);
     const unnamed = people.filter((person) => !person.name);
+    if (areHidden)
+      return (
+        [...named, ...unnamed]
+          // with thumbnails
+          .filter((person) => !!person.thumbnailPath)
+          .filter((person) => !person.isHidden)
```

Comment:
You can merge these two versions if you just do !areHidden or !person.isHidden here.

Reply:
Although, maybe it is supposed to be with hidden or is not hidden. Also you may need to use a transform to change the query param to a boolean. I think it is just a string right now. There is a toBoolean method we use other places.

Reply:
Yep saw it, I will use it.

Reply:
Why does this endpoint does not need a transform to change the query param to a boolean ?

https://github.com/immich-app/immich/blob/9012cf6946d71bde1fe06d1f3fb76b978f6e2af9/server/src/immich/controllers/user.controller.ts#L42-L44

Reply:
It might be a bug. We probably only send true, which even as a string is truthy. There might be a bug that sending false doesn't work. Other places use a dto class for the query params and then use the transformer.

Reply:
Ok good to know, I was using this endpoint as an example